crates/sparrow-session/src/session.rs

```diff
@@ -25,6 +25,13 @@ pub struct Session {
     data_context: DataContext,
     dfg: Dfg,
     key_hash_inverse: HashMap<GroupId, Arc<ThreadSafeKeyHashInverse>>,
+    /// Keeps track of the uuid mapping.
+    ///
+    /// This is a bit of a hack that allows to serialize the uuid instead of the
+    /// `Arc<dyn Udf` into the compute plan, then look up that mapping during plan
+    /// construction. Once we run on multiple machines we'll need to serialize the
+    /// udf.
+    udfs: HashMap<Uuid, Arc<dyn Udf>>,
```

Comment:
Even when we do, I'd expect us to serialize a map from trait to UDF -- specifically, the same UDF may appear multiple times, and we wouldn't want to pickle it each time it occurs. Using the UUID for indirection is a reasonable pattern for serializations.